bb8/src/api.rs

```diff
@@ -547,3 +555,36 @@ impl<E> ErrorSink<E> for NopErrorSink {
         Box::new(*self)
     }
 }
+
+/// Error type returned by `Pool::add(conn)`
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub enum AddError<C> {
+    /// The connection was broken before it could be added.
+    Broken(C),
+    /// Unable to add the connection to the pool due to insufficient capacity.
+    NoCapacity(C),
+}
+
+impl<E> fmt::Display for AddError<E>
+where
+    E: error::Error + 'static,
```

Comment:
Nit: please put these bounds into the `impl` declaration for conciseness (so `impl<E: error::Error + 'static>`).